src_firmware/src/storage.c

```diff
@@ -67,91 +86,233 @@ uint8_t get_template_size(void)
 	return alloc_size;
 }
 
-void set_template(struct Template new_template)
+int set_template(struct Template new_template)
 {
-	struct Template tmp_template;
-	int idx;
+	if (!initialized) {
+		return ERR_NOT_INITIALISED;
+	}
 
-	idx = get_template_by_name(new_template.name, &tmp_template);
-	if (idx >= 0) {
-		(void)nvs_write(&fs, idx + TEMPLATES_START_ID,
+	struct Template tmp_template;
+	uint8_t idx;
+	int error_get_template;
+	int error_write;
+
+	// Get template index from name.
+	error_get_template = get_template_and_id_by_name(new_template.name, &tmp_template, &idx);
+	if (error_get_template == SUCCESS) { // if index exists:
+		error_write = nvs_write(&fs, idx + TEMPLATES_START_ID, // write new template
 				&new_template, sizeof(struct Template));
+		if (error_write < 0) {
+			handle_errno_error(error_write, __LINE__);
```

Comment:
note: Should we really handle line at which error occurred? It seems like an overkill that can be useful only at first stages of debugging. Implementing a proper logger (not shell/printk) would be better here. We can do it someday later

Reply:
there are functions where there are multiple functions called which can result in error from errno, therefore it is good to know where the error occured. Later it could be wrapped with `#ifdef DEBUG` or somethig like that

Reply:
Or just logged at error occurrence with zephyr LOG macros. But that can be done later